src/server/ua_services_attribute.c

```diff
@@ -253,11 +253,14 @@ readValueAttributeComplete(UA_Server *server, UA_Session *session,
                 break;
             }
             /* Set the result */
-            if(rangeptr)
-                return UA_Variant_copyRange(
-                    (const UA_Variant *)&vn->valueBackend.backend.external.value,
-                    &v->value, *rangeptr);
             retval = UA_DataValue_copy(*vn->valueBackend.backend.external.value, v);
```

Comment:
This can be inefficient.
Making a full copy just to clear the variant.
You could instead make a “shallow” copy of the DataValue and _init the variant portion before calling copyrange.

Reply:
Thanks for the comments. Now pushed on 1.3 branch and made a copyVariantRange function for the DataValue type.